lib/HttpClient.js

```diff
@@ -306,36 +288,32 @@ function rawRequest(opts, cb) {
                 requestTimer = null;
 
                 var err = errors.createRequestTimeoutErr(opts, req);
-                dtrace._rstfy_probes['client-error'].fire(function () {
-                    return ([id, err.toString()]);
-                });
-
-                cb(err, req);
-
-                if (req) {
-                    req.abort();
-                }
-                process.nextTick(function () {
-                    emitResult(err, req, null);
-                    emitAfter(req, null, err);
-                });
+                req._forcedAbortErr = err;
+                req.abort();
```

Comment:
nit/thinking out loud: do we want to `assert` on `req` here since it seems we rely on it being a request object? I'm thinking about it more from a code documentation perspective.